src/inference/Inference.jl

```diff
@@ -189,10 +189,10 @@ function AbstractMCMC.sample(
     kwargs...
 )
     if resume_from === nothing
-        return AbstractMCMC.mcmcsample(rng, model, sampler, N;
-                                       chain_type=chain_type, progress=progress, kwargs...)
+        return AbstractMCMC.mcmcsample(rng, model, sampler, N + min(div(N, 10), 1_000);
```

Comment:
This means users will get more than the requested `N` samples - I think this should be reverted?

Reply:
Sure, we have to take into account the adaption steps - but changing this line here means that a call such as `sample(model, NUTS(), 10)` will return not 10 but 1010 samples? These discarded steps only have to be specified as keyword argument but not added to the positional argument (number of samples). AbstractMCMC will added `discard_initial` to these internally to the requested number of samples automatically.